core/src/test/scala/org/apache/spark/ExternalShuffleServiceSuite.scala

```diff
@@ -255,4 +255,22 @@ class ExternalShuffleServiceSuite extends ShuffleSuite with BeforeAndAfterAll wi
       }
     }
   }
+
+  test("SPARK-38640: using external shuffle service fetching disk persisted blocks") {
+    val confWithRddFetchEnabled = conf.clone.set(config.SHUFFLE_SERVICE_FETCH_RDD_ENABLED, true)
```

Comment:
Shall we test with both `true` and `false`?

Reply:
I guess if you want to make sure just having the shuffle service enabled doesn't affect this. I expect there's other tests testing all the default configs with all the caching behaviors

Reply:
Testing both enabled and disabled